pkg/crds/names.go

```diff
@@ -0,0 +1,300 @@
+
+// RequiredCRDs returns a list of CRD to install based on enabled features.
+func RequiredCRDs() []string {
+	requiredCRDS := BasicCRDs()
+	if features.ProvisioningV2.Enabled() {
+		requiredCRDS = append(requiredCRDS, ProvisioningV2CRDs()...)
+		if features.RKE2.Enabled() {
+			requiredCRDS = append(requiredCRDS, RKE2CRDs()...)
+		}
+		if features.EmbeddedClusterAPI.Enabled() {
+			requiredCRDS = append(requiredCRDS, CAPICRDs()...)
+		}
+		if features.Fleet.Enabled() {
+			requiredCRDS = append(requiredCRDS, "managedcharts.management.cattle.io")
+		}
+	}
+	if features.Fleet.Enabled() {
+		requiredCRDS = append(requiredCRDS, FleetCRDs()...)
+	}
+	if features.MCM.Enabled() {
+		requiredCRDS = append(requiredCRDS, MCMCRDs()...)
+	}
+	if features.Auth.Enabled() {
+		requiredCRDS = append(requiredCRDS, AuthCRDs()...)
+	}
+	if features.MonitoringV1.Enabled() {
+		requiredCRDS = append(requiredCRDS, MonitoringV1CRDs()...)
+	}
+	return requiredCRDS
+}
+
+// BasicCRDs returns a list of CRD names needed to run rancher.
+func BasicCRDs() []string {
+	return []string{
+		"apps.catalog.cattle.io",
+		"clusterrepos.catalog.cattle.io",
+		"operations.catalog.cattle.io",
+		"apiservices.management.cattle.io",
+		"clusters.management.cattle.io",
+		"clusterregistrationtokens.management.cattle.io",
+		"features.management.cattle.io",
+		"podsecurityadmissionconfigurationtemplates.management.cattle.io",
+		"preferences.management.cattle.io",
+		"settings.management.cattle.io",
+		"navlinks.ui.cattle.io",
+	}
+}
+
+// ProvisioningV2CRDs returns a list of CRD names needed for ProvisioningV2.
+func ProvisioningV2CRDs() []string {
+	return []string{
+		"clusters.provisioning.cattle.io",
+	}
+}
+
+// CAPICRDs returns a list of CRD names needed for CAPI.
+func CAPICRDs() []string {
+	return []string{
+		"machines.cluster.x-k8s.io",
+		"machinehealthchecks.cluster.x-k8s.io",
+		"machinedeployments.cluster.x-k8s.io",
+		"machinesets.cluster.x-k8s.io",
+		"clusters.cluster.x-k8s.io",
+	}
+}
+
+// RKE2CRDs returns a list of CRD names needed for RKE2.
+func RKE2CRDs() []string {
+	return []string{
+		"clusters.provisioning.cattle.io",
+		"custommachines.rke.cattle.io",
+		"etcdsnapshots.rke.cattle.io",
+		"rkebootstraps.rke.cattle.io",
+		"rkebootstraptemplates.rke.cattle.io",
+		"rkeclusters.rke.cattle.io",
+		"rkecontrolplanes.rke.cattle.io",
+	}
+}
+
+// FleetCRDs returns a list of CRD names needed for Fleet.
+func FleetCRDs() []string {
+	return append(
+		bootstrapFleet(),
+		"fleetworkspaces.management.cattle.io",
+	)
+}
+
+func bootstrapFleet() []string {
+	return []string{
+		"bundles.fleet.cattle.io",
+		"clusters.fleet.cattle.io",
+		"clustergroups.fleet.cattle.io",
+	}
+}
+
+// AuthCRDs returns a list of CRD names needed for authentication.
+func AuthCRDs() []string {
+	return []string{
+		"authconfigs.management.cattle.io",
+		"groups.management.cattle.io",
+		"groupmembers.management.cattle.io",
+		"tokens.management.cattle.io",
```

Comment:
This looks like it's duplicated in MCMCRDs, is that intentional?

Reply:
Yes, they can be installed via two independent paths
https://github.com/rancher/rancher/blob/release/v2.7/pkg/api/norman/server/managementstored/setup.go#L80-L143
 https://github.com/rancher/rancher/blob/release/v2.7/pkg/auth/api/setup.go#L63-L70